components/Admin/Employees/EmployeeCreation.tsx

```diff
@@ -0,0 +1,186 @@
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useCallback, useEffect, useState } from "react";
+import { Employee } from "../../../modules/employee/employee.interface";
+import employeeService from "../../../services/employee/employee.service";
+
+export default function EmployeeCreation() {
+  const router = useRouter();
+  const user = useUser();
+  const supabase = useSupabaseClient();
+  const [employee, setEmployee] = useState<Employee>(null);
+  const [isFetching, setIsFetching] = useState<boolean>(true);
+  
+  const { id } = router.query as { id: string | null };
```

Comment:
Better put the type inside an interface to clearly define what is expected from the Router Query here.